File: app/src/main/java/net/veldor/flibustaloader/receivers/BookLoadedReceiver.java
package net.veldor.flibustaloader.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;
import android.widget.Toast;

import net.veldor.flibustaloader.notificatons.Notificator;

public class BookLoadedReceiver extends BroadcastReceiver {

    public static final String EXTRA_BOOK_NAME = "book name";
    public static final String EXTRA_ACTION_TYPE = "action type";
    public static final String EXTRA_BOOK_TYPE = "book mime";

    public static final String ACTION_TYPE_OPEN = "open";
    public static final String ACTION_TYPE_SHARE = "share";
    public static final String EXTRA_AUTHOR_FOLDER = "author folder";
    public static final String EXTRA_SEQUENCE_FOLDER = "sequence folder";
    public static final String EXTRA_RESERVED_SEQUENCE_FOLDER = "reserved sequence folder";


    @Override
    public void onReceive(Context context, Intent intent) {
        String name = intent.getStringExtra(EXTRA_BOOK_NAME);
        String type = intent.getStringExtra(EXTRA_BOOK_TYPE);
        Log.d("surprise", "BookLoadedReceiver onReceive " + type);
        Notificator.getInstance().sendLoadedBookNotification(name, type);
        Toast.makeText(context, name + " - загружено!", Toast.LENGTH_LONG).show();
    }
}

File: app/src/main/java/net/veldor/flibustaloader/receivers/BookLoadedReceiver.kt
package net.veldor.flibustaloader.receivers

import android.content.Intent
import android.widget.Toast
import android.content.BroadcastReceiver
import android.content.Context
import android.util.Log
import net.veldor.flibustaloader.notificatons.NotificationHandler

class BookLoadedReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val name = intent.getStringExtra(EXTRA_BOOK_NAME)
        val type = intent.getStringExtra(EXTRA_BOOK_TYPE)
        Log.d("surprise", "BookLoadedReceiver onReceive $type")
        NotificationHandler.instance.sendLoadedBookNotification(name, type)
        Toast.makeText(context, "$name - загружено!", Toast.LENGTH_LONG).show()
    }

    companion object {
        const val EXTRA_BOOK_NAME = "book name"
        const val EXTRA_ACTION_TYPE = "action type"
        const val EXTRA_BOOK_TYPE = "book mime"
        const val ACTION_TYPE_OPEN = "open"
        const val ACTION_TYPE_SHARE = "share"
        const val EXTRA_AUTHOR_FOLDER = "author folder"
        const val EXTRA_SEQUENCE_FOLDER = "sequence folder"
        const val EXTRA_RESERVED_SEQUENCE_FOLDER = "reserved sequence folder"
    }
}